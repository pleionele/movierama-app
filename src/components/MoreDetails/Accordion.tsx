import React from 'react';

function AccordionItem() {}
