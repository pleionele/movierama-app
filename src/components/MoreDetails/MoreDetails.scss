// html,
// body,
// div + div {
//   width: 100vw;
//   height: 100vh;
// }

// div + div {
//   list-style: none;
//   position: relative;
//   display: flex;
//   flex-direction: row;
//   flex-wrap: nowrap;
//   align-items: stretch;
//   overflow: hidden;
// }

// article div {
//   text-align: left;
//   width: 58vw;
// }

// div + div:hover article {
//   flex: initial;
//   width: 10%;
// }

// div + div article:hover {
//   width: 60%;
// }

// article > div {
//   opacity: 0;
//   transition: opacity 0.2s ease 0;
// }

// div + div article:hover > div {
//   opacity: 1;
//   transition: opacity 0.3s ease 0.3s;
// }

// @media (max-width: 900px) {
//   div + div article {
//     padding: 2vh 3vw;
//   }

//   div + div article > h2 {
//     transform: rotate(90deg);
//     bottom: 23vh;
//     min-width: 12em;
//     text-align: left;
//     transform: rotate(-90deg);
//     transform-origin: 0 0 0;
//     opacity: 1;
//   }

//   div + div article:hover > h2 {
//     opacity: 0;
//   }

//   article div {
//     max-height: calc(72%);
//     overflow-y: auto;
//   }
// }

header {
  display: flex;
  justify-content: center;
  align-content: center;
  padding: 15px;
  background-color: #3c3c3c;
  // border: 3px solid #f43c3c;
}

header h1 {
  color: #eee;
  font-size: 20px;
  font-weight: 700;
  text-transform: uppercase;
}
.acr__container {
  width: 100%;
  max-width: 768%;
  margin: 0 auto;
}
.acr__item {
  margin: 15px;
  padding: 15px;
  background-color: white;
  border-radius: 8px;
  box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.2);

  .open &__header {
    margin-bottom: 10px;
  }
  &__header {
    position: relative;
    font-size: 20px;
    padding-right: 80px;
    transition: all 0.4s ease;
  }

  &__header::after {
    content: '';
    position: absolute;
    top: 50%;
    right: 0px;
    transform: translateY(-50%);
    width: 30px;
    height: 30px;
    transition: all 0.4s ease;
  }

  .open &__body {
    max-height: 1000px;
    opacity: 1;
  }
  &__body {
    opacity: 0;
    max-height: 0;
    overflow-y: hidden;
    transition: all 0.6s ease;

    iframe {
      /* Mobile Styles */
      @media only screen and (max-width: 400px) {
        max-width: 300px !important;
      }
    }
  }
}
