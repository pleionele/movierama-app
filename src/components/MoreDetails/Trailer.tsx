import * as React from 'react';
import YouTube from 'react-youtube';

export const Trailer = (props: any) => {
  const { data } = props;
  try {
    return (
      <YouTube
        videoId={data.video[0].key}
        containerClassName="acr__item__body"
        className="acr__item__body"
      />
    );
  } catch (e) {
    return <div className="acr__item__body"> No available trailer</div>;
  }
};
