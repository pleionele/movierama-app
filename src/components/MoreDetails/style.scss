html,
body,
div + div {
  width: 100vw;
  height: 100vh;
}

div + div {
  list-style: none;
  position: relative;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: stretch;
  overflow: hidden;
}

article div {
  text-align: left;
  width: 58vw;
}

div + div:hover article {
  flex: initial;
  width: 10%;
}

div + div article:hover {
  width: 60%;
}

article > div {
  opacity: 0;
  transition: opacity 0.2s ease 0;
}

div + div article:hover > div {
  opacity: 1;
  transition: opacity 0.3s ease 0.3s;
}

@media (max-width: 900px) {
  div + div article {
    padding: 2vh 3vw;
  }

  div + div article > h2 {
    transform: rotate(90deg);
    bottom: 23vh;
    min-width: 12em;
    text-align: left;
    transform: rotate(-90deg);
    transform-origin: 0 0 0;
    opacity: 1;
  }

  div + div article:hover > h2 {
    opacity: 0;
  }

  article div {
    max-height: calc(72%);
    overflow-y: auto;
  }
}
