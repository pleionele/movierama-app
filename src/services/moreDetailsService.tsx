import { getMovieReviews } from '../api/get-movie-reviews';
import { getMovieSimilar } from '../api/get-movie-similar';
import { getMovieVideo } from '../api/get-movie-video';

export const moreDetailsService = (id: number) => {
  const idToString = id.toString();
  const reviews: any = getMovieReviews(idToString);
  const similarMovies: any = getMovieSimilar(idToString);
  const video: any = getMovieVideo(idToString);

  return Promise.all([reviews, similarMovies, video]).then(results => {
    return Object.assign(results[0], results[1], results[2]);
  });
};
