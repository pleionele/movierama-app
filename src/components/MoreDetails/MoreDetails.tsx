import React, { Fragment, useEffect, useState } from 'react';
import { moreDetailsService } from '../../services/moreDetailsService';
import './MoreDetails.scss';
import ReviewList from '../Review/ReviewList';
import YouTube from 'react-youtube';
import { SimilarMoviesList } from '../SimilarMovies/SimilarMovies';

interface MoreDetailsProps {
  movieId: number;
  // closePopup: () => void;
}
const MoreDetails: React.FC<MoreDetailsProps> = props => {
  const [data, setData] = useState();
  const [isloading, setLoading] = useState(true);
  const [isDisplaying, setDisplay] = useState(true);
  const [accordion, setAccordion] = useState([
    {
      id: 0,
      open: false,
    },
    {
      id: 1,
      open: false,
    },
    {
      id: 2,
      open: false,
    },
  ]);

  const toggleTabs = (index: number) => {
    setAccordion(
      accordion.map((item, i) => {
        if (i === index) {
          item.open = !item.open;
        } else {
          item.open = false;
        }
        return item;
      })
    );
  };
  const toggleTrailer = () => {
    toggleTabs(0);
  };

  const toggleReviews = () => {
    toggleTabs(1);
  };

  const toggleSimilarMovies = () => {
    toggleTabs(2);
  };

  useEffect(() => {
    (() => {
      let returnedData: any = null;
      moreDetailsService(props.movieId)
        .then(results => {
          returnedData = results;
        })
        // tslint:disable-next-line: no-console
        .catch(error => console.log(error))
        .finally(() => {
          setData(returnedData);
          setLoading(false);
        });
    })();
  }, []);
  // TODO Check if video exists

  return (
    <Fragment>
      {isDisplaying && (
        <div>
          {isloading && <div> Please wait..</div>}
          {!isloading && (
            <section>
              <header id="acr-label">More details</header>
              <div className="acr__container">
                <div
                  className={'acr__item ' + (accordion[0].open ? 'open' : '')}
                  onClick={toggleTrailer}
                  data-testid="youtubeTrailer"
                >
                  <div className="acr__item__header">Trailer</div>
                  <YouTube
                    videoId={data.video[0].key}
                    containerClassName="acr__item__body"
                    className="acr__item__body"
                  />
                </div>
                <div
                  onClick={toggleReviews}
                  className={'acr__item ' + (accordion[1].open ? 'open' : '')}
                  data-testid="reviewsList"
                >
                  <div className="acr__item__header">Reviews</div>
                  <div className="acr__item__body">
                    <ReviewList reviews={data.reviews} />
                  </div>
                </div>
                <div
                  onClick={toggleSimilarMovies}
                  className={'acr__item ' + (accordion[2].open ? 'open' : '')}
                  data-testid="similarMovies"
                >
                  <div className="acr__item__header">Similar Movies</div>
                  <div className="acr__item__body">
                    <SimilarMoviesList movies={data.similarMovies} />
                  </div>
                </div>
              </div>
            </section>
          )}
        </div>
      )}
    </Fragment>
  );
};

export default MoreDetails;
