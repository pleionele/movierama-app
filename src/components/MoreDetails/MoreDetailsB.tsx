import React, { useState, useEffect, Children } from 'react';
import { moreDetailsService } from '../../services/moreDetailsService';
import { Config } from '../../config/default';
import './style.scss';
import ReviewList from '../Review/ReviewList';
import YouTube from 'react-youtube';
import { MovieItem } from '../Movie/Movie';

// ● Video Trailer (if any)
// ● Reviews (up to 2)
// ● Similar Movies
interface MoreDetailsProps {
  movieId: number;
  closePopup: () => void;
}
const MoreDetails: React.FC<MoreDetailsProps> = props => {
  const [data, setData] = useState();
  const [isloading, setLoading] = useState(true);
  const [isDisplaying, setDisplay] = useState(true);
  const [accordion, setAccordion] = useState([
    {
      id: 0,
      open: false,
    },
    {
      id: 1,
      open: false,
    },
  ]);

  const toggleTabs = (index: number) => {
    setAccordion(
      accordion.map((item, i) => {
        console.log('in here ', item, i);

        if (i === index) {
          item.open = !item.open;
        } else {
          item.open = false;
        }
        return item;
      })
    );
  };

  useEffect(() => {
    (() => {
      let returnedData: any = null;
      moreDetailsService(props.movieId)
        .then(results => {
          console.log(results);
          returnedData = results;
        })
        .catch(error => console.log(error))
        .finally(() => {
          setData(returnedData);
          setLoading(false);
        });
    })();
  }, []);

  return (
    <React.Fragment>
      {isDisplaying && (
        <div>
          {isloading && <div> Please wait..</div>}
          {!isloading && (
            <section
            // aria-labelledby="acr-label" aria-hidden="true"
            >
              <header id="acr-label">More details</header>
              <div className="acr__container">
                <div
                  className={'acr__item ' + (accordion[0].open ? 'open' : '')}
                  onClick={e => {
                    e.stopPropagation();
                    console.log('aaa');
                    toggleTabs(0);
                  }}
                >
                  <div className="acr__item__header">Trailer</div>
                  {/* <YouTube videoId={data?.video[0].key} /> */}
                  <div className="acr__item__body">Hello1</div>
                </div>
                <div
                  onClick={e => {
                    e.stopPropagation();
                    console.log('bbb');
                    toggleTabs(1);
                  }}
                  className={'acr__item ' + (accordion[1].open ? 'open' : '')}
                >
                  <div className="acr__item__header">Reviews</div>
                  {/* <YouTube videoId={data?.video[0].key} /> */}
                  <div className="acr__item__body">Hello12</div>
                </div>
              </div>
            </section>
          )}
        </div>
      )}
    </React.Fragment>
  );
};

export default MoreDetails;
