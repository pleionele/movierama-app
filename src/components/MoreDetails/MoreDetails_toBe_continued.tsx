import React, { useState, useEffect } from 'react';
import { moreDetailsService } from '../../services/moreDetailsService';
import { Config } from '../../config/default';
import './style.scss';
import ReviewList from '../Review/ReviewList';
import YouTube from 'react-youtube';

// ● Video Trailer (if any)
// ● Reviews (up to 2)
// ● Similar Movies
interface MoreDetailsProps {
  movieId: number;
  closePopup: () => void;
}
const MoreDetails: React.FC<MoreDetailsProps> = props => {
  const [data, setData] = useState();
  const [isloading, setLoading] = useState(true);
  const [active, isactive] = useState(false);
  useEffect(() => {
    (() => {
      let returnedData: any = null;
      moreDetailsService(props.movieId)
        .then(results => {
          console.log(results);
          returnedData = results;
        })
        .catch(error => console.log(error))
        .finally(() => {
          setData(returnedData);
          setLoading(false);
        });
    })();
  }, []);

  return (
    <React.Fragment>
      <div>
        {isloading && <div> Please wait..</div>}
        {!isloading && (
          <section
            className="acr--show"
            id="acr-text"
            tabIndex={-1}
            role="dialog"
            aria-labelledby="acr-label"
            aria-hidden="true"
          >
            <header id="acr-label">More details</header>
            <div className="acr_trailer-container">
              <a>Trailer</a>
              <div>
                <YouTube videoId={data?.video[0].key} />
              </div>
            </div>
            <div className="acr_reviews">
              <a>Reviews</a>
              <ReviewList reviews={data?.reviews} />
            </div>
            <div className="acr_similar">
              <a>Similar Movies</a>
              <ReviewList reviews={data?.reviews} />
            </div>
          </section>
        )}
      </div>
    </React.Fragment>
  );
};

export default MoreDetails;
