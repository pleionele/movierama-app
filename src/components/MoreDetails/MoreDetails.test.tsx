import * as React from 'react';
import MoreDetails from './MoreDetails';
import {
  render,
  waitForElement,
  getByTestId,
  wait,
} from '@testing-library/react';
import { moreDetailsService } from '../../services/moreDetailsService';
// @ts-ignore
import * as payloadMoreDetails from '../../api/stubs/payload-more-details.json';
import { act } from 'react-dom/test-utils';

jest.mock('../../services/moreDetailsService', () => {
  return {
    moreDetailsService: jest.fn(),
  };
});
describe('More Details', () => {
  beforeEach(() => {
    (moreDetailsService as jest.Mock).mockReturnValue(
      Promise.resolve(payloadMoreDetails)
    );
  });

  it('should match the snapshot', () => {
    act(async () => {
      const { asFragment } = render(<MoreDetails movieId={323} />);
      Promise.resolve();
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
