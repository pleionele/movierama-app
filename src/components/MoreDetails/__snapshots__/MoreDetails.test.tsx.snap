// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`More Details should match the snapshot 1`] = `
<DocumentFragment>
  <div>
     Please wait..
  </div>
</DocumentFragment>
`;
